### PR TITLE
Add channel QoS setting

### LINKED ISSUE #11: Need a way to set consumer prefetch
After using this library, I am facing the limitation of prefetching. I used to do that for amqp library by using `basicQos` in channel level.
Basically, I think without this feature, current throughput could be not improved in my consumer side.

### PR DESCRIPTION
**Pull request by acogoluegnes**
Fixes #11

### CODE REVIEW

File: src/main/java/com/rabbitmq/jms/client/RMQConnection.java [hunk 1]
```diff
@@ -82,6 +84,13 @@
     private volatile boolean canSetClientID = true;
 
     /**
+     * QoS setting for channels created by this connection.
+     *
+     * @see com.rabbitmq.client.Channel#basicQos(int)
+     */
+    private final int channelsQos;
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/jms/client/RMQConnection.java]: Does this value have to be `final`? QoS changes on the fly are not very common but possible in AMQP 0-9-1.

File: src/main/java/com/rabbitmq/jms/client/RMQConnection.java [hunk 2]
```diff
@@ -47,7 +47,9 @@
  */
 public class RMQConnection implements Connection, QueueConnection, TopicConnection {
 
-    private final Logger logger = LoggerFactory.getLogger(RMQConnection.class);;
+    public static final int NO_CHANNEL_QOS = -1;
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/jms/client/RMQConnection.java]: Is the intent here to have no limit by default? Then it should be `0`.

File: src/main/java/com/rabbitmq/jms/client/RMQConnection.java (hunk 1 above)
Comment (acogoluegnes) [inline src/main/java/com/rabbitmq/jms/client/RMQConnection.java]: It doesn't have to be `final`. The intent is to have a `RMQConnection` creating AMQP channels with the same settings, not more. Not sure users are coding against the `RMQConnection` class, but more against the JMS `*Connection` interfaces. We can make it `volatile` and expose it with getter/setter, it should be harmless.

Comment (michaelklishin) [inline src/main/java/com/rabbitmq/jms/client/RMQConnection.java]: OK, this is not how Java client works but we can safely switch it to per-session (connection) strategy later if JMS users care enough. Going the other way would be a breaking change.

**Pull request merged**